### ISSUE TITLE
Help needed

### ISSUE THREAD

**Issue by xjtuqihan** [IBM/DPFS#31]
Could you please tell me how to integrate dpfs and snap.

**Comment by Peter-JanGootzen** [comment]
What is not clear about the instruction in the README? Do you have access to SNAP?

Side-note:
DPFS has become outdated, NVIDIA has released the DOCA DevEmu virtio-fs API in DOCA, which should be used instead.
https://docs.nvidia.com/doca/sdk/doca+devemu+virtio-fs/index.html

**Comment by xjtuqihan** [comment]
Yes, i can access to Snap. "Using in SNAP is currently only possible with a prototype firmware and some alterations to the SNAP library. You can reach out to us on how to integrate DPFS and SNAP." After reading this from readme I want to know how to patch snap. I need to config it right? And after that, I need to compile and execute dpfs_fuse, dpfs_hal, and one of the backends, right?
I find DevEmu virtio-fs need bf3. So I want to use dpfs still.

**Comment by Peter-JanGootzen** [comment]
Regarding the prototype firmware, if you run `sudo mst start; sudo mlxconfig -d /dev/mst/${INSERT_BF2_PCICONF} -e q`, do you see the virtio-fs features? If yes, then great you can turn them on :) If not, then you need to contact your Nvidia representative (for example your contact that gave you SNAP access) and ask access for a firmware that has the virtio-fs features.

Regarding patching SNAP, you will have to figure that out for yourself because of NDA reasons, given the instructions in the README, you should be able to find it quite easily.

After compiling everything you only need to execute one of the backends, it starts up the other layers for you.

DPFS on the BF2 was a research prototype, it has several issues. If you want to make serious use of this stack you need to consider BF3.
